### ISSUE TITLE
hasBody() No Longer Supports containsSubstring()

### ISSUE THREAD

**Issue by commonsguy** [http4k/http4k#167]
```kotlin
    response shouldMatch hasStatus(Status.OK)
      .and(hasBody(containsSubstring("Um, hi!")))
      .and(hasHeader("Content-type", "text/html; charset=UTF-8"))
```

Given that `response` is a `Response`, the above code works with http4k 3.34.3 and fails to compile starting with 3.35.0:

```
None of the following functions can be called with the arguments supplied: 
@JvmName public fun hasBody(expected: Matcher<String?>): Matcher<HttpMessage> defined in org.http4k.hamkrest
public fun hasBody(expected: Matcher<Body>): Matcher<HttpMessage> defined in org.http4k.hamkrest
public fun hasBody(expected: CharSequence): Matcher<HttpMessage> defined in org.http4k.hamkrest
public fun hasBody(expected: Regex): Matcher<HttpMessage> defined in org.http4k.hamkrest\
```

`containsSubstring()` in hamkrest appears to return a `Matcher<CharSequence>`, which I guess is incompatible with `Matcher<String?>`.

I'm not sure if this was a planned regression or not, but it would be cool if `containsSubstring()` were a valid `Matcher` to use with `hasBody()`.

Let me know if you need additional information &mdash; thanks for considering this!

**Comment by s4nchez** [comment]
Hi @commonsguy,

We didn't plan to retire `hasBody(expected: Matcher<String>)`. I'll put it back and add a test using `containsSubstring` to avoid that from happening again.

Thank you for reporting this!

**Comment by s4nchez** [comment]
@commonsguy That should be available on version 3.35.2.

(going through the pipeline as we speak)

**Comment by daviddenton** [comment]
@commonsguy Just FYI - we're re-releasing 3.35.2 as 3.36.0 - there was some incubator code that crept into the core module that might cause name confusion, so please use the new version instead. It's building now.

**Comment by commonsguy** [comment]
I can confirm that this is fixed in 3.36.1 &mdash; many thanks!
